modules/apps/commerce/commerce-order-web/src/main/java/com/liferay/commerce/order/web/internal/frontend/CommerceOrderCreateDateClayTableDataSetFilter.java

```diff
@@ -0,0 +1,170 @@
+import com.liferay.frontend.taglib.clay.data.set.filter.ClayDataSetFilter;
+import com.liferay.frontend.taglib.clay.data.set.filter.DateClayDataSetFilterItem;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.model.User;
+import com.liferay.portal.kernel.search.BaseModelSearchResult;
+import com.liferay.portal.kernel.search.Field;
+import com.liferay.portal.kernel.search.QueryConfig;
+import com.liferay.portal.kernel.search.SearchContext;
+import com.liferay.portal.kernel.search.Sort;
+import com.liferay.portal.kernel.search.SortFactoryUtil;
+import com.liferay.portal.kernel.service.UserService;
+
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.stream.Stream;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Luca Pellizzon
+ */
+@Component(
+	enabled = false, immediate = true,
+	property = "clay.data.set.display.name=" + CommerceOrderDataSetConstants.COMMERCE_DATA_SET_KEY_ALL_ORDERS,
+	service = ClayDataSetFilter.class
+)
+public class CommerceOrderCreateDateClayTableDataSetFilter
+	extends BaseDateRangeClayDataSetFilter {
+
+	@Override
+	public String getId() {
+		return "createDate";
+	}
+
+	@Override
+	public String getLabel() {
+		return "order-create-date-range";
+	}
+
+	public DateClayDataSetFilterItem getMax() {
+		Calendar calendar = Calendar.getInstance();
+
+		return new DateClayDataSetFilterItem(
+			calendar.get(Calendar.DAY_OF_MONTH),
+			calendar.get(Calendar.MONTH) + 1, calendar.get(Calendar.YEAR));
+	}
+
+	public DateClayDataSetFilterItem getMin() {
+		Calendar calendar = Calendar.getInstance();
+
+		int day = calendar.get(Calendar.DAY_OF_MONTH);
+		int month = calendar.get(Calendar.MONTH);
+		int year = calendar.get(Calendar.YEAR) - 1;
+
+		try {
+			User currentUser = _userService.getCurrentUser();
+
+			SearchContext searchContext = _buildSearchContext(
+				currentUser.getCompanyId(), 0, 1);
+
+			BaseModelSearchResult<CommerceOrder>
+				commerceOrderBaseModelSearchResult =
+					_commerceOrderLocalService.searchCommerceOrders(
+						searchContext);
+
+			List<CommerceOrder> commerceOrders =
+				commerceOrderBaseModelSearchResult.getBaseModels();
+
+			if (!commerceOrders.isEmpty()) {
+				CommerceOrder commerceOrder = commerceOrders.get(0);
+
+				Date createDate = commerceOrder.getCreateDate();
+
+				calendar.setTime(createDate);
+
+				day = calendar.get(Calendar.DAY_OF_MONTH);
+				month = calendar.get(Calendar.MONTH) + 1;
+				year = calendar.get(Calendar.YEAR);
+			}
+		}
+		catch (PortalException portalException) {
+			if (_log.isInfoEnabled()) {
+				_log.info("No order found", portalException);
+			}
+		}
+
+		return new DateClayDataSetFilterItem(day, month, year);
+	}
+
+	@Override
+	public String getType() {
+		return "dateRange";
+	}
```

Comment:
@luca-pellizzon maybe it's better to implement the getType() in the abstract class... so if we need to have another filter for data ranges not for orders we don't need to reimplement this method

Reply:
thanks, I just forgot to move it inside BaseDateRangeClayDataSetFilter, will do it